### PR TITLE
Use flex instead Grid in Galley component

### PR DESCRIPTION
**Pull request by JuanAlmeida12**
#### What is the purpose of this pull request?

Use flex instead grid in `Gallery` and spaces between `GalleryItems` can be customized.

#### How should this be manually tested?

[Access this workspace](https://galleryspace--storecomponents.myvtex.com/)

#### Screenshots or example usage
galleryspace--storecomponents myvtex com_ 2
galleryspace--storecomponents myvtex com_ 3

#### Types of changes

* [ ] Bug fix (a non-breaking change which fixes an issue)
* [ ] New feature (a non-breaking change which adds functionality)
* [x] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/Gallery.js [hunk 1]
```diff
@@ -29,20 +49,28 @@ class Gallery extends Component {
...
     )
   }
+
+  getItemsPerRow = () => {
+    const { maxItemsPerRow, gap, minItemWidth, width } = this.props
+    let maxItems = maxItemsPerRow
+    while ((minItemWidth + gap) * maxItems >= width) {
+      --maxItems
+    }
+    return maxItems
```
Comment (GustavoMA) [inline react/Gallery.js]: Isn't `maxItems = width / (minItemWidth + gap)` ?

File: react/searchResult.css [hunk 2]
```diff
@@ -80,20 +80,9 @@
...
-  grid-gap: 0.375rem;
-  margin: 0 auto;
-}
+.gallery {}
 
-.galleryTwoColumns {
-  display: grid;
-  grid-template-columns: repeat(2, 1fr);
-}
+.galleryTwoColumns {}
```
Comment (GustavoMA) [inline react/searchResult.css]: This is not used anymore, you can delete it :)

File: react/Gallery.js (hunk 1 above)
Comment (JuanAlmeida12) [reply react/Gallery.js]: Really is better :)

File: react/Gallery.js [hunk 3]
```diff
@@ -13,13 +14,32 @@ import searchResult from './searchResult.css'
 /**
  * Canonical gallery that displays a list of given products.
  */
+
+const TWO_ITEMS = 2
```
Comment (theomoura) [inline react/Gallery.js]: Could you improve those variable names? Maybe put `TWO_ROW_ITEMS` or something alike for better readability.

File: react/Gallery.js [hunk 4]
```diff
@@ -13,13 +14,32 @@ import searchResult from './searchResult.css'
...
+const ONE_ITEM = 1
+
 class Gallery extends Component {
+  constructor(props) {
+    super(props)
+    this.state = { prevLayoutMode: this.props.layoutMode }
```
Comment (theomoura) [inline react/Gallery.js]: I think the use of the constructor here is pointless, you can set the state propertly more directly, what do you think?

File: react/Gallery.js [hunk 5]
```diff
@@ -29,20 +49,25 @@ class Gallery extends Component {
       </div>
     )
   }
+
+  get itemsPerRow() {
+    const { maxItemsPerRow, gap, minItemWidth, width } = this.props
+    const maxItems = Math.floor(width/(minItemWidth + gap))
```
Comment (theomoura) [inline react/Gallery.js]: Put spaces between `width / (minItemWidth`

**Pull request merged**